docs/developers/customize-karmada-scheduler.md

```diff
@@ -0,0 +1,191 @@
+
+```
+.
+├── apienablement
+├── clusteraffinity
+├── clusterlocality
+├── spreadconstraint
+├── tainttoleration
+├── testfilter
+│ ├── test_filter.go
+```
+
+The content of the test_filter.go file is as follows, and the specific filtering logic implementation is hidden.
+
+```go
+package testfilter
+
+import (
+	"context"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/scheduler/framework"
+)
+
+const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	Name = "TestFilter"
+)
+
+type TestFilter struct{}
+
+var _ framework.FilterPlugin = &TestFilter{}
+
+// New instantiates the APIEnablement plugin.
+func New() (framework.Plugin, error) {
+	return &TestFilter{}, nil
+}
+
+// Name returns the plugin name.
+func (p *TestFilter) Name() string {
+	return Name
+}
+
+// Filter checks if the API(CRD) of the resource is enabled or installed in the target cluster.
+func (p *TestFilter) Filter(ctx context.Context, placement *policyv1alpha1.Placement,
+	bindingSpec *workv1alpha2.ResourceBindingSpec, cluster *clusterv1alpha1.Cluster) *framework.Result {
+
+	// implementation
+
+	return framework.NewResult(framework.Success)
+}
+```
+
+For a filter plugin, you must implement `framework.FilterPlugin` interface. And for a score plugin, you must implement `framework.ScorePlugin` interface.
+
+## Register the plugin
+
+Edit the [cmd/scheduler/main.go](https://github.com/karmada-io/karmada/blob/master/cmd/scheduler/main.go):
+
+```go
+package main
+
+import (
+	"os"
+
+	"k8s.io/component-base/cli"
+	_ "k8s.io/component-base/logs/json/register" // for JSON log format registration
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	_ "sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"github.com/karmada-io/karmada/cmd/scheduler/app"
+	"github.com/karmada-io/karmada/pkg/scheduler/framework/plugins/testfilter"
+)
+
+func main() {
+	stopChan := controllerruntime.SetupSignalHandler().Done()
+	command := app.NewSchedulerCommand(stopChan, app.WithPlugin(testfilter.Name, testfilter.New))
+	code := cli.Run(command)
+	os.Exit(code)
+}
+
+```
+
+## Package the scheduler
+
+After you register the plugin, you need to package your scheduler binary into a container image.
+
+```shell
+cd karmada
+make image-karmada-scheduler
+```
+
+```shell
+kubectl --kubeconfig ~/.kube/karmada.config --context karmada-host edit deploy/karmada-scheduler -nkarmada-system
+...
+ spec:
+      automountServiceAccountToken: false
+      containers:
+      - command:
+        - /bin/karmada-scheduler
+        - --kubeconfig=/etc/kubeconfig
+        - --bind-address=0.0.0.0
+        - --secure-port=10351
+        - --enable-scheduler-estimator=true
+        - --v=4
+        image: docker.io/karmada/karmada-scheduler:xxxxx
```

Comment:
ditto. Add a line to describe how to use the new image.

After building the image, users might need to load it to a registry, obviously not `docker.io/karmada/karmada-scheduler`.